src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -209,6 +215,231 @@ Scalar<DataType> time_deriv_of_lapse(
+tnsr::iJ<DataType, SpatialDim, Frame> spatial_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  tnsr::iJ<DataType, SpatialDim, Frame> deriv_shift {};
+  GeneralizedHarmonic::spatial_deriv_of_shift<SpatialDim, Frame, DataType>(
+    make_not_null(&deriv_shift), lapse, inverse_spacetime_metric,
+    spacetime_unit_normal, phi);
+  return deriv_shift;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void time_deriv_of_shift(
+    const gsl::not_null<tnsr::I<DataType, SpatialDim, Frame>*> dt_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t a = 0; a < SpatialDim + 1; ++a) {
+        // first term
+        dt_shift->get(i) -= get(lapse) * pi.get(j+1, a)
+            * spacetime_unit_normal.get(a) * inverse_spatial_metric.get(i, j);
+        // second term
+        for (size_t k = 0; k < SpatialDim; ++k) {
+          dt_shift->get(i) += shift.get(j)
+              * spacetime_unit_normal.get(a) * phi.get(j, k+1, a)
+              * inverse_spatial_metric.get(i, k);
+        }
+      }
+    }
+    dt_shift->get(i) *=  get(lapse);
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::I<DataType, SpatialDim, Frame> time_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  auto dt_shift =
+    make_with_value<tnsr::I<DataType, SpatialDim, Frame>>(get(lapse), 0.);
+  GeneralizedHarmonic::time_deriv_of_shift<SpatialDim, Frame, DataType>(
+    make_not_null(&dt_shift), lapse, shift, inverse_spatial_metric,
+    spacetime_unit_normal, phi, pi);
+  return dt_shift;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void time_deriv_of_lower_shift(
+    const gsl::not_null<tnsr::i<DataType, SpatialDim, Frame>*> dt_lower_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
```

Comment:
It seems strange to me that this function takes both `pi` and `dt_spatial_metric`, since `dt_spatial_metric` is a very simple function of `pi`, `phi`, `lapse`, and `shift`, all of which are passed in here.  Why not compute dt_spatial_metric in place in this function? Even more worrying is whether the results of the `dt_spatial_metric` function from #1179 will be passed in here; I think that could ruin hyperbolicity.

Reply:
I agree. This is also discussed here: https://github.com/sxs-collaboration/spectre/pull/1172#discussion_r235464820

Reply:
I think we want to use a DataBox value (passed in argument) as long as it cannot affect the hyperbolicity of the system. If it can then I think we should take any minor hit on extracting time derivatives from `pi` locally in order to guarantee the system remains at least strongly hyperbolic.

Reply:
@markscheel  -- I agree completely and my intention was that this function should ultimately be calling `GeneralizedHarmonic::time_deriv_of_spatial_metric` to get `dt_spatial_metric` from GH variables alone (setting `gamma1 = -1`). However, since `time_deriv_of_spatial_metric` is part of another PR  (#1179 ) I had left `dt_spatial_metric` here as an argument (with the tacit understanding, I guess, that this `dt_spatial_metric` comes from GH vars only).  I can see that that was confusing, and so I've modified the code to not take in `dt_spatial_metric` at all but extract it from `pi, phi, lapse, shift`, and have also put in an explanation that this computation should eventually be replaced with a call to `time_deriv_of_spatial_metric`.